tfhe-zk-pok/src/proofs/pke_v2.rs

```diff
@@ -328,8 +328,14 @@ pub fn prove<G: Curve>(
 
     let PrivateCommit { r, e1, m, e2, .. } = private_commit;
 
-    assert!(c2.len() <= k);
-    let k = k.min(c2.len());
+    assert!(c2.len() <= k_max);
+    let k = k_max.min(c2.len());
```

Comment:
with the above assert this can be just `let k = c2.len()`

Reply:
indeed, we could do k = c2.len(); k <= k_max